R/corridor.R

```diff
@@ -135,56 +139,74 @@ corridor_end_points <- function(river, network, aoi = NULL) {
+corridor_end_points <- function(river_network, spatial_network) {
+  # Only consider the edges of the spatial network
+  spat_network_edges <- sf::st_geometry(sf::st_as_sf(spatial_network, "edges"))
 
   # Find intersections with the spatial network, and add these points as nodes
   # to the river network. These nodes are the candidates corridor end points
   river_network_edges <- sf::st_geometry(sf::st_as_sf(river_network, "edges"))
-  intersections <- sf::st_intersection(river_network_edges, spat_net_edges)
+  intersections <- sf::st_intersection(river_network_edges, spat_network_edges)
+  # Make sure they are "POINTS" (no "MULTIPOINTS")
+  intersections <- sfheaders::sfc_cast(intersections, "POINT")
   river_network <- sfnetworks::st_network_blend(river_network, intersections)
   nodes <- nearest_node(river_network, intersections)
 
-  # Pick a reference point on the selected river segment, either up- or down-
-  # stream from all crossings. Pick this as the closest node to the aoi boundary
-  reference <- nearest_node(river_network, sf::st_boundary(aoi))
-
-  # Compute the network distance from the reference node to all candidate end
-  # points. The closest and furthest nodes are selected as corridor end points
+  # Compute the network distance between all the candidate end points. The ones
+  # that are furthest away from each others are selected as corridor end points
   river_network <- add_weights(river_network, weight_name = "weight")
-  distances <- sfnetworks::st_network_cost(river_network, from = reference,
+  distances <- sfnetworks::st_network_cost(river_network, from = nodes,
                                            to = nodes, weights = "weight")
-  end_points <- c(nodes[which.min(distances[1, ])],
-                  nodes[which.max(distances[1, ])])
+
+  indices <- which(distances == max(distances), arr.ind = TRUE)[1, ]
+  end_points <- c(nodes[indices["row"]], nodes[indices["col"]])
   if (end_points[1] == end_points[2]) {
     stop("Corridor start- and end-points coincide!")
   }
   end_points
 }
 
-#' Split the area of interest (AoI) by a river.
+#' Draw the regions corresponding to the two river banks
 #'
-#' Return the fragments produced. If more than two fragments are obtained, only
-#' return the two largest fragments.
+#' These are constructed as single-sided buffers around the river geometry (see
+#' [`river_buffer()`] for the implementation and refinement steps).
 #'
-#' @param bbox Bounding box defining the extent of the area of interest
-#' @param river A simple feature geometry representing the river centerline
-#'
-#' @return A simple feature geometry set of two areas of interest
-#' @importFrom rlang .data
-split_aoi <- function(bbox, river) {
-  regions <- split_by(as_sfc(bbox), river)
+#' @param river River spatial features provided as a [`sfnetworks::sfnetwork`]
+#'   or [`sf::sf`]/[`sf::sfc`] object.
+#' @param width (Approximate) width of the regions
```

Comment:
Why approximate? I am just curious.

Reply:
Actually, this the "real" width - it  is`max_width` in `delineate_corridor` to be an approximate indication of the maximum width of the corridor, since the actual corridor can extend slightly outside the region if the edges of the network "spill" outside it. I will the text here, thanks for checking this scrupulously!